examples/ex-tmp.cpp

```diff
@@ -0,0 +1,420 @@
+//                                MFEM Example 22
```

Comment:
These example files should be cleaned up and renumbered to ex24 (there is ex23 already)

Reply:
Thank you for the heads up. @cjvogl ex_tmp is ex23 but a non-smart renaming from ex22. ex-tmp is not part of the final changes.

Reply:
This is what I figured.  Given this, I also advocate removing `examples/ex-tmp.cpp` from the file tree and renaming `examples/ex-23.cpp` to `examples/ex-24.cpp` (with appropriate changes to the comments within and from the checklist)

Reply:
> These example files should be cleaned up and renumbered to ex24 (there is ex23 already)

I am sorry, I got confused with another PR, the example number should be indeed 23 not 24... I'll fix it.